python.d/haproxy.chart.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+# Description: haproxy netdata python.d module
+# Author: Pawel Krupa (paulfantom)
+
+from base import SocketService, UrlService
+import csv
+
+# default module values (can be overridden per job in `config`)
+# update_every = 2
+priority = 60000
+retries = 60
+
+# charts order (can be overridden if you want less charts, or different order)
+ORDER = ['qcur', 'scur', 'bin', 'bout']
+POSITION = [2, 4, 8, 9]
+
+CHARTS = {
+    'qcur': {
+        'options': ["", "Current queue", 'per sec', '', '', 'line'],
+        'lines': [
+            ['name', None, 'incremental']
+        ]},
+    'scur': {
+        'options': ["", "Current session rate", 'per sec', '', '', 'line'],
+        'lines': [
+            ['name', None, 'incremental']
+        ]},
+    'bin': {
+        'options': ["", "Bytes in", 'kilobytes/s', '', '', 'line'],
+        'lines': [
+            ['name', None, 'incremental', 1, 1024]
+        ]},
+    'bout': {
+        'options': ["", "Bytes out", 'kilobytes/s', '', '', 'line'],
+        'lines': [
+            ['name', None, 'incremental', 1, 1024]
+        ]},
+}
+
+
+class Service(SocketService, UrlService):
+    def __init__(self, configuration=None, name=None):
+        self.use_socket = None
+        if 'unix_socket' in configuration:
```

Comment:
You could assign a little less.  Consider:

    self.use_socket = 'unix_socket' in configuration
    if self.use_socket: ...

lets you get rid of assigning to `use_socket` three times.